Sig.App.Backend/BackgroundJobs/AddingFundToCard.cs

```diff
@@ -187,7 +187,7 @@ public async Task AddFundToSpecificBeneficiary(Id beneficiaryId, BeneficiaryType
                 .Where(x => x.Id == subscriptionIdLong).FirstAsync();
 
             var beneficiary = await db.Beneficiaries
-                .Include(x => x.Card).ThenInclude(x => x.Transactions)
+                .Include(x => x.Card).ThenInclude(x => x.Transactions).ThenInclude(x => (x as SubscriptionAddingFundTransaction).SubscriptionType)
```

Comment:
Ça ne plante pas ça si le bénéficiaire a des transactions qui ne sont pas des SubscriptionAddingFundTransaction?

Ça fait quoi dans le fond? Ça fait un genre de _where type of transaction is SubscriptionAddingFundTransaction_ ?

Reply:
Le OfType est effectivement un genre de Where Type == <monType>

https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.oftype?view=net-8.0

Reply:
Même si la méthode utilisé est ThenInclude?

Reply:
La ligne qui plantait était la suivante

var subscriptionAddedFundCount = beneficiary.Card.Transactions.OfType<SubscriptionAddingFundTransaction>().Count(x => x.SubscriptionType.SubscriptionId == subscription.Id);

Parce que je loadais pas le SubscriptionType dans le cas qu'une transaction est de type SubscriptionAddingFundTransaction. Donc en loadant le SubscriptionType pour les SubscriptionAddingFundTransaction, j'ai toutes l'info que j'ai besoin :)